### PR TITLE
Add computeLedgerHash method

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

**Comment by clark800** [comment]
Hold off on merging this one, I need to do parsing on the new ledger header format

**Comment by clark800** [comment]
OK ready

File: src/core/ledger.js [hunk 1]
```diff
@@ -156,4 +157,24 @@ Ledger.prototype.calc_account_hash = function(options) {
   return account_map.hash();
 };
 
+// see rippled Ledger::updateHash()
+Ledger.calculateLedgerHash =
+Ledger.prototype.calculateLedgerHash = function(ledgerHeader) {
+  const so = new SerializedObject();
+  const prefix = 0x4C575200;
+  const totalCoins = (new BigNumber(ledgerHeader.total_coins)).toString(16);
```
Comment (sublimator) [inline src/core/ledger.js]: you could use `bn.js ` to create a BN and just pass that directly to Int64.serialize?  It's what that routine does anyway with hex strings

Comment (clark800) [inline src/core/ledger.js]: I think it's more valuable to stick to one bignumber library throughout the whole api directory.

File: src/api/offline/ledgerhash.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    accepted: header.accepted,
+    closed: header.closed,
+    account_hash: header.accountHash,
+    close_time: header.closeTime,
+    close_time_resolution: header.closeTimeResolution,
+    close_flags: header.closeFlags,
+    hash: header.ledgerHash,
+    ledger_hash: header.ledgerHash,
+    ledger_index: header.ledgerVersion.toString(),
+    seqNum: header.ledgerVersion.toString(),
```
Comment (lumberj) [inline src/api/offline/ledgerhash.js]: Why are some keys camelcased and some underscored?

File: src/api/offline/ledgerhash.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    parent_close_time: header.parentCloseTime,
+    total_coins: header.totalDrops,
+    totalCoins: header.totalDrops,
+    transaction_hash: header.transactionHash
+  };
+}
+
+function hashLedgerHeader(ledgerHeader) {
+  const header = convertLedgerHeader(ledgerHeader);
+  return common.core.Ledger.calculateLedgerHash(header);
```
Comment (lumberj) [inline src/api/offline/ledgerhash.js]: is `calculateLedgerHash` part of the legacy `core`?

File: src/api/offline/ledgerhash.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+function computeLedgerHash(ledgerHeader: Object, transactions: Array<Object>) {
```
Comment (lumberj) [inline src/api/offline/ledgerhash.js]: Add a return type annotation?

File: src/api/offline/ledgerhash.js (hunk 2 above)
Comment (clark800) [inline src/api/offline/ledgerhash.js]: rippled

File: src/api/offline/ledgerhash.js (hunk 3 above)
Comment (clark800) [inline src/api/offline/ledgerhash.js]: It was added to core, see ledger.js below.

File: src/api/offline/ledgerhash.js (hunk 4 above)
Comment (clark800) [inline src/api/offline/ledgerhash.js]: ok

**Comment by clark800** [comment]
@lumberj updated

**Comment by lumberj** [comment]
LGTM

**Pull request merged**